mrp_food/models/mrp_bom.py

```diff
@@ -0,0 +1,194 @@
+                        # Next letter is space, but it's not the end of the name
+                        if bom.product_id.name[i + 1] == " ":
+                            i += 1
+                            continue
+                        # Loop with next letter
+                        else:
+                            new_tla_end = bom.product_id.name[i + 1].upper()
+                            new_tla = new_tla_start + new_tla_end
+                            i += 1
+                    bom.tla = new_tla
+
+    @api.multi
+    def _get_bom_code_start(self):
+        self.ensure_one()
+        if not self.product_id:
+            return 0
+        else:
+            bom_code = self._BOM_CODE_SEQ_START
+            if self.env.user.company_id.code:
+                bom_code += ("-") + self.env.user.company_id.code
+            bom_code += ("-") + self.tla
+            return bom_code
+
+    # Override function
+    # We have to override copy function, otherwise, when we duplicate BoM
+    # the bom_count of the product is not accurate (it counts the one being created)
+    @api.multi
+    def copy(self, default=None):
+        default = dict(default or {})
+        if self.product_id:
+            count = self.product_id.bom_count
+        new_code = self._get_bom_code_start() + ("-") + str(count + 1)
+        default.update(
+            {
+                "code": new_code,
+            }
+        )
+        return super(MrpBom, self).copy(default)
+
+    @api.depends("product_id")
+    def _compute_proposal_code(self):
+        for bom in self:
+            if bom.product_id:
+                bom.code = self._get_bom_code_start()
+                bom.code += ("-") + str(self.product_id.bom_count + 1)
+
+    def _inverse_proposal_code(self):
+        return True
+
+    #
+    # Other functions
+    #
+    @api.multi
+    @api.depends("product_id", "bom_line_ids.allergen_ids")
+    def _compute_bom_allergen_ids(self):
+        for bom in self:
+            bom.bom_allergen_ids = bom.product_id.allergen_ids
+            for bom_line in bom.bom_line_ids:
+                bom.bom_allergen_ids = [x.id for x in bom_line.product_id.allergen_ids]
+
+    @api.multi
+    @api.depends("product_id", "bom_line_ids")
+    def _compute_standard_price_total(self):
+        for bom in self:
+            bom.standard_price_total = sum(
+                x.standard_price_subtotal for x in bom.bom_line_ids
+            )
+
+    @api.multi
+    @api.depends("product_id", "bom_season_ids", "bom_line_ids")
+    def _compute_seasonal(self):
+        for bom in self:
+            # Handling BoM Seasonalities
+            # One Tag is in season, and we considere the BoM in season
+            today = fields.Date.today()
+            for seasonality in bom.bom_season_ids:
+                for period in seasonality.seasonality_line_ids:
+                    if period.date_start <= today <= period.date_end:
+                        bom.is_bom_seasonal = True
```

Comment:
for period in bom.mapped("bom_season_ids.seasonality_line_ids"):
   if period.date_start <= today <= period.date_end:
       bom.is_bom_seasonal = True
       break

Reply:
:heavy_check_mark: